### PR TITLE
Indicate the beginning of executed scripts

### PR DESCRIPTION
**Pull request by ChengTian**
parent https://github.com/aspnet/dnx/issues/1764

@balachir @muratg @davidfowl 

Sample output:
capture

### CODE REVIEW

**Comment by muratg** [comment]
Looks good, though I wonder "in project.json" part is useful or not. I assume there won't be other DNX pieces which will execute scripts.   I think we can check-in like this, and revisit later.

:shipit:

File: src/Microsoft.Framework.PackageManager/Scripts/ScriptExecutor.cs [hunk 1]
```diff
@@ -23,6 +30,8 @@ public bool Execute(Runtime.Project project, string scriptName, Func<string, str
                 return true;
             }
 
+            Report.WriteLine($"Executing script '{scriptName}' in {Runtime.Project.ProjectFileName}");
```
Comment (dougbu) [inline src/Microsoft.Framework.PackageManager/Scripts/ScriptExecutor.cs]: `Report` can be `null`. Need to use `Report?.WriteLine(...)`

File: src/Microsoft.Framework.PackageManager/Scripts/ScriptExecutor.cs [hunk 2]
```diff
@@ -15,6 +15,13 @@ public class ScriptExecutor
     {
         private static readonly string ErrorMessageTemplate = "The '{0}' script failed with status code {1}.";
 
+        public ScriptExecutor(IReport report = null)
+        {
+            Report = report;
+        }
+
+        public IReport Report { get; set; }
```
Comment (dougbu) [inline src/Microsoft.Framework.PackageManager/Scripts/ScriptExecutor.cs]: Curious: Why does this have a setter, let alone a `public` setter?

Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Scripts/ScriptExecutor.cs]: To make it work with `RestoreCommand` class, we need this public setter. `RestoreCommand` cannot finalize the value of `Reports` when it is constructing `ScriptExecutor`, so setting `IReport` with constructor is not enough.

We may need to redesign `RestoreCommand` to get rid of this but it's a separate topic.

File: src/Microsoft.Framework.PackageManager/Scripts/ScriptExecutor.cs (hunk 1 above)
Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Scripts/ScriptExecutor.cs]: Good point

**Pull request merged**